workflow/controller/taskresult.go

```diff
@@ -55,25 +56,29 @@ func (woc *wfOperationCtx) taskResultReconciliation() {
 	for _, obj := range objs {
 		result := obj.(*wfv1.WorkflowTaskResult)
 		nodeID := result.Name
-		old := woc.wf.Status.Nodes[nodeID]
-		new := old.DeepCopy()
+		old, err := woc.wf.Status.Nodes.Get(nodeID)
+		if err != nil {
+			woc.log.Fatalf("was unable to obtain node for %s", nodeID)
+			panic(fmt.Sprintf("unable to obtain node for %s", nodeID))
```

Comment:
panic seems extreme here, continue instead?

Reply:
I bubbled up an error instead, I think this is safer than continuing.